### ISSUE TITLE
20

### ISSUE THREAD

**Issue by halj3317-bit** [halj3317-bit/2#9]
import pandas as pd
import numpy as np
from sklearn.preprocessing import OneHotEncoder, StandardScaler
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import r2_score

df = pd.read_csv("data/day20_integration.csv")

X = df.drop("price", axis=1)
y = df["price"]

X = pd.get_dummies(X)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

model = LinearRegression()
model.fit(X_train, y_train)
print("Baseline R2:", r2_score(y_test, model.predict(X_test)))

X["interaction"] = X.iloc[:,0] * X.iloc[:,1]
X["log_feature"] = np.log1p(X.iloc[:,0])

scaler = StandardScaler()
X = pd.DataFrame(scaler.fit_transform(X), columns=X.columns)

X_train2, X_test2, y_train2, y_test2 = train_test_split(X, y, test_size=0.2, random_state=42)

model2 = LinearRegression()
model2.fit(X_train2, y_train2)
print("Engineered R2:", r2_score(y_test2, model2.predict(X_test2)))
